plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -805,13 +805,17 @@ private String buildImage(final DockerConnection dockerConnection,
                         progressMonitor.updateProgress(progressStatus);
                     }
 
-                    throw new DockerException("Docker image build failed. Image id not found in build output.", 500);
+                    throw new ImageNotFoundException("Docker image build failed. Image id not found in build output.");
                 });
 
                 return imageIdFuture.get();
             } catch (ExecutionException e) {
                 // unwrap exception thrown by task with .getCause()
-                throw new DockerException(e.getCause().getLocalizedMessage(), 500);
+                if (e.getCause() instanceof ImageNotFoundException) {
+                    throw new ImageNotFoundException(e.getCause().getLocalizedMessage());
```

Comment:
`e.getCause()` returns `Throwable`, buildImage() method can not throw such exception